app/src/main/java/net/squanchy/schedule/ScheduleService.kt

```diff
@@ -35,68 +25,23 @@ class FirestoreScheduleService(
     override fun schedule(onlyFavorites: Boolean): Observable<Schedule> {
         return Observable.combineLatest(dbService.scheduleView(), dbService.timezone(), combineInAPair())
             .map { pagesAndTimeZone ->
-                val (schedulePages, timeZone) = pagesAndTimeZone
 
+                val (schedulePages, timeZone) = pagesAndTimeZone
                 schedulePages.map { schedulePage ->
                     SchedulePage(
                         schedulePage.day.id,
                         LocalDate(schedulePage.day.date),
-                        schedulePage.events.map { it.toEvent(schedulePage.day.id, timeZone) }
+                        // TODO pass the favourites here instead of null to know which talks are favourites
+                        schedulePage.events.map { it.toEvent(checksum, timeZone, null) }
                             .sortedBy { it.startTime }
                             .filterOnlyFavorites(onlyFavorites)
                     )
                 }
             }
-            .map { Schedule(it) }
+            .map(::Schedule)
     }
 
-    private fun <T, U> combineInAPair(): BiFunction<List<T>, U, Pair<List<T>, U>> =
-        BiFunction { schedulePages, timeZone -> Pair(schedulePages, timeZone) }
-
-    private fun FirestoreEvent.toEvent(dayId: String, timeZone: DateTimeZone) = Event(
-        id,
-        checksum.getChecksumOf(id),
-        LocalDateTime(startTime),
-        LocalDateTime(endTime),
-        title,
-        place.toPlace(),
-        track.toTrack(),
-        speakers.toSpeakersList(checksum),
-        ExperienceLevel.tryParsingFrom(experienceLevel),
-        dayId,
-        Event.Type.fromRawType(type),
-        false, // TODO fetch favourites
-        description.optional(),
-        timeZone
-    )
-
-    private fun FirestoreTrack?.toTrack() = optional().map {
-        Track(
-            it.id,
-            it.name,
-            it.accentColor.optional(),
-            it.textColor.optional(),
-            it.iconUrl.optional()
-        )
-    }
-
-    private fun FirestorePlace?.toPlace() = optional().map {
-        Place(it.id, it.name, it.floor.optional())
-    }
-
-    private fun List<FirestoreSpeaker>.toSpeakersList(checksum: Checksum) = map {
-        Speaker(
-            checksum.getChecksumOf(it.id),
-            it.id,
-            it.name,
-            it.bio,
-            it.companyName.optional(),
-            it.companyUrl.optional(),
-            it.personalUrl.optional(),
-            it.photoUrl.optional(),
-            it.twitterUsername.optional()
-        )
-    }
+    private fun <T, U> combineInAPair(): BiFunction<List<T>, U, Pair<List<T>, U>> = BiFunction(::Pair)
```

Comment:
can you inline this now? or does it lose the types?

Reply:
Yeah it does lose the type